junit-platform-launcher/src/main/java/org/junit/platform/launcher/core/DefaultLauncherSession.java

```diff
@@ -73,11 +83,17 @@ LauncherSessionListener getListener() {
 	public void close() {
 		if (launcher.delegate != ClosedLauncher.INSTANCE) {
 			launcher.delegate = ClosedLauncher.INSTANCE;
+			store.close();
 			listener.launcherSessionClosed(this);
```

Comment:
To allow the listener to read things from the `store`, we should first call the listener and then close the store.